docs/glossary.rst

```diff
@@ -0,0 +1,274 @@
+
+        Workers are configured and launched through the `CPI`. Data and requests
+        for data are provided to the worker by its client. The worker is able
+        to publish both final results and intermediate results for the work it
+        is executing. More generally, the worker publishes updates to the work
+        graph, which include state updates for existing graph nodes as well as
+        new nodes. New nodes are necessary to hold static data, to describe
+        operations dispatched in support of higher level operations, or to
+        extend the work graph (such as in support of adaptive work flows).
+
+        A worker may be launched to perform a single task, to perform a sequence
+        of tasks, or to participate in an :term:`ensemble`.
+
+    CPI
+        .. todo:: Define CPI.
+
+    Context
+      Abstraction for the entity that maps work to a computing environment.
+      Instances may be long-lived and participate in owning/managing work and
+      data references.
+
+    Session
+      Abstraction for the entity representing work that is executing on resources
+      allocated by an instance of a Context implementation. The Session is the
+      scoped active state of a Context while computing resources are held.
+
+.. topic:: *Context* and *Session*
+
+    roughly map to terms like *Executor* and *Task* in some other frameworks.
+    Distinctions relate to the lifetime of the :term:`Context` instance, and the fact that
+    it owns both the work specification (including operation and data handles)
+    and the computing resources.
+    The :term:`Context` instance owns resources (on behalf of the client) that may
+    otherwise be owned directly by the client, and so its lifetime must span all
+    references to resources, operation handles, and data futures.
+
+.. glossary::
+
+    discovered task
+        A task that has become runnable, but was not already scheduled.
+        This term is intentionally vague as the requirements and constraints of
+        work management are explored. The primary usage refers to a task that
+        has been generated due to adaptations in the work flow. It may also
+        apply to tasks that may be scheduled opportunistically, or simply to
+        the change of state when a task's input dependencies have been met.
+
+.. _user classification:
+
+User classes and characteristics
+--------------------------------
+
+Users are assumed to be molecular science researchers using Python scripts to
+express and execute simulation and analysis work consisting of multiple
+simulation and analysis tasks, using software tools from multiple packages.
+
+Software tools are individually accessible as Python modules or as command line
+tools.
+
+Computational work may require multiple invocations (multiple HPC jobs) to complete.
+
+The following classes of user are not necessarily mutually exclusive.
+
+.. glossary::
+
+    basic user
```

Comment:
@mrshirts writes:
>  A researcher writing a Python script to control standard software." Maybe solely "call existing SCALE-MS functions operating on existing SCALE-MS objects?" since "standard software" is not defined. Though I would probably call this an intermediate user - a basic user would only run existing SCALE-MS workflows, and only change the data, instructions, and parameters fed into that workflow.